File: ui/ide-extensions/phase2/eclipse/src/com/voltnuerongrid/eclipse/client/VngSecretStore.java
package com.voltnuerongrid.eclipse.client;

import org.eclipse.equinox.security.storage.ISecurePreferences;
import org.eclipse.equinox.security.storage.SecurePreferencesFactory;
import org.eclipse.equinox.security.storage.StorageException;

/**
 * D-5: Eclipse secret storage for the VoltNueronGrid admin key, backed by the
 * platform {@link ISecurePreferences} (OS keychain / encrypted secure storage).
 * The admin key is never persisted in plain preferences.
 */
public final class VngSecretStore {

    private static final String NODE = "com.voltnuerongrid.eclipse";
    private static final String KEY_ADMIN = "adminKey";

    private VngSecretStore() {
    }

    /** Store the admin key encrypted in Eclipse secure storage. */
    public static void putAdminKey(String value) throws StorageException {
        ISecurePreferences node = SecurePreferencesFactory.getDefault().node(NODE);
        node.put(KEY_ADMIN, value == null ? "" : value, true);
    }

    /** Retrieve the admin key from secure storage, or empty when unset. */
    public static String getAdminKey() {
        try {
            ISecurePreferences node = SecurePreferencesFactory.getDefault().node(NODE);
            return node.get(KEY_ADMIN, "");
        } catch (StorageException e) {
            return "";
        }
    }
}

File: ui/ide-extensions/phase2/jetbrains/src/main/kotlin/com/voltnuerongrid/jetbrains/client/VngSecretStore.kt
package com.voltnuerongrid.jetbrains.client

import com.intellij.credentialStore.CredentialAttributes
import com.intellij.credentialStore.Credentials
import com.intellij.credentialStore.generateServiceName
import com.intellij.ide.passwordSafe.PasswordSafe

/**
 * D-5: JetBrains secret storage for the VoltNueronGrid admin key, backed by the
 * IDE [PasswordSafe] (OS keychain / encrypted credential store). The key is
 * never written to the plain settings XML.
 */
object VngSecretStore {

    private fun attributes(): CredentialAttributes =
        CredentialAttributes(generateServiceName("VoltNueronGrid", "adminKey"))

    /** Persist the admin key in the IDE PasswordSafe. */
    fun putAdminKey(value: String) {
        PasswordSafe.instance.set(attributes(), Credentials("admin", value))
    }

    /** Retrieve the admin key from the IDE PasswordSafe, or empty when unset. */
    fun getAdminKey(): String =
        PasswordSafe.instance.getPassword(attributes()).orEmpty()
}
